neo/Helper.cs

```diff
@@ -245,6 +250,20 @@ internal static long WeightedAverage<T>(this IEnumerable<T> source, Func<T, long
             return sum_value / sum_weight;
         }
 
+        internal static BigInteger WeightedAverage<T>(this IEnumerable<T> source, Func<T, BigInteger> valueSelector, Func<T, BigInteger> weightSelector)
+        {
+            BigInteger sum_weight = BigInteger.Zero;
+            BigInteger sum_value = BigInteger.Zero;
+            foreach (T item in source)
+            {
+                BigInteger weight = weightSelector(item);
+                sum_weight += weight;
+                sum_value += valueSelector(item) * weight;
+            }
+            if (sum_value == BigInteger.Zero) return BigInteger.Zero;
```

Comment:
```suggestion
            if (sum_value == BigInteger.Zero || sum_weight == BigInteger.Zero) return BigInteger.Zero;
```

Reply:
If `sum_weight == 0` but `sum_value != 0`, then an exception should be thrown.